irods/test/data_obj_test.py

```diff
@@ -931,69 +931,81 @@ def test_replica_number(self):
-        obj_path = '/{zone}/home/{username}/{filename}'.format(**locals())
-
-        # make test file
-        obj_content = u'foobar'
-        checksum = base64.b64encode(hashlib.sha256(obj_content.encode('utf-8')).digest()).decode()
-        with open(test_file, 'w') as f:
-            f.write(obj_content)
-
-        # put test file onto default resource
-        options = {kw.REG_CHKSUM_KW: ''}
-        session.data_objects.put(test_file, obj_path, **options)
-
-        # make ufs resources and replicate object
-        ufs_resources = []
-        for i in range(number_of_replicas):
-            resource_name = unique_name(my_function_name(),i)
-            resource_type = 'unixfilesystem'
-            resource_host = session.host
-            resource_path = '/tmp/{}'.format(resource_name)
-            ufs_resources.append(session.resources.create(
-                resource_name, resource_type, resource_host, resource_path))
-
-            session.data_objects.replicate(obj_path, resource=resource_name)
+                session.data_objects.replicate(obj_path, resource=resource_name)
 
-        # refresh object
-        obj = session.data_objects.get(obj_path)
+            # refresh object
+            obj = session.data_objects.get(obj_path)
 
-        # verify each replica's checksum
-        for replica in obj.replicas:
-            self.assertEqual(replica.checksum, 'sha2:{}'.format(checksum))
+            # verify each replica's checksum
+            for replica in obj.replicas:
+                self.assertEqual(replica.checksum, 'sha2:{}'.format(checksum))
 
-        # now repave test file
-        obj_content = u'bar'
-        checksum = base64.b64encode(hashlib.sha256(obj_content.encode('utf-8')).digest()).decode()
-        with open(test_file, 'w') as f:
-            f.write(obj_content)
+            # now repave test file
+            obj_content = u'bar'
+            checksum = base64.b64encode(hashlib.sha256(obj_content.encode('utf-8')).digest()).decode()
+            with open(test_file, 'w') as f:
+                f.write(obj_content)
 
-        # update all replicas
-        options = {kw.REG_CHKSUM_KW: '', kw.ALL_KW: ''}
-        session.data_objects.put(test_file, obj_path, **options)
-        obj = session.data_objects.get(obj_path)
+            # update all replicas
+            # -------------------
+            # Note this doesn't work in iRODS 4.3.0 and later because necessary options to overwrite all
+            # replicas of a checksummed object would include VERIFY_CHKSUM_KW and FORCE_FLAG_KW, which in com-
+            # bination with the REG_CHKSUM_KW and ALL_KW produces an USER_INCOMPATIBLE_PARAMS error.  If we
+            # choose to omit VERIFY_CHKSUM_KW, we get SYS_NOT_ALLOWED with a warning that VERIFY_CHKSUM_KW is
+            # required.  So it seems the repave operation performed by this test would require a put followed
+            # by a REPL with the full set of options required to repave.
```

Comment:
We are skipping this test for any version >4.2.11, but this comment states that this doesn't work in 4.3.0. Does this work for pre-4.2.12 (i.e. 4-2-stable)? I'm guessing that this should say "iRODS versions later than 4.2.11".

Is this a bug in the server? The last sentence suggests that this may be more of a TODO comment, but I find this concerning.

Reply:
oh wait that commit is already here, so we're not skipping this test at all now.